### ISSUE TITLE
Sunrise and sunset indications on images

### ISSUE THREAD

**Issue by atruskie** [QutEcoacoustics/audio-analysis#250]
## Is your feature request related to a problem? Please describe.

We've previously displayed sunrise and sunset times on FCS images. However, we removed that code because it was hardcoded to work with the Gympie region and constantly broke.

## Describe the solution you'd like

To see a sundial plot added to visualization images.

## Additional context

We'd need some consistent way to read location metadata for the input files. 

Suggested by @ThomasLuypaertGitHub

**Comment by towsey** [comment]
What would a sundial plot look like in this instance if it is to be added to a rectangular image? We previously used a 10-15 pixel high track at the bottom the 24-hr spectrogram to illustrate sunlight hours. This does not cover the spectrogram and adds little to image height. It also has the added advantage that it can be overlaid with other temporal information, such as rain, cloud, wind info if these become available.

**Comment by atruskie** [comment]
Yeah, _sundial_ was just the term I used to describe the sunlight plot you previously had. 

My intention was to recreate what you previously had, but more generic and reusable.

**Comment by Akash10987** [comment]
> 
> 
> What would a sundial plot look like in this instance if it is to be added to a rectangular image? We previously used a 10-15 pixel high track at the bottom the 24-hr spectrogram to illustrate sunlight hours. This does not cover the spectrogram and adds little to image height. It also has the added advantage that it can be overlaid with other temporal information, such as rain, cloud, wind info if these become available.

How I can add Sunrise/Sunset time and Moon phase information in a 24-hr spectrogram as described by you in the comment?

**Comment by towsey** [comment]
A sundial plot could be added back but it would require the user to supply sunrise and sunset information since these are dependent on longitude of the recording. I am assuming recording metadata will not necessarily supply this info. Moon phase info could be added based on date.
Question is, are we overloading the ldfc spectrogram generation command with information?

**Comment by Akash10987** [comment]
> 
> 
> A sundial plot could be added back but it would require the user to supply sunrise and sunset information since these are dependent on longitude of the recording. I am assuming recording metadata will not necessarily supply this info. Moon phase info could be added based on date.
> Question is, are we overloading the ldfc spectrogram generation command with information?

I have sunrise/sunset and moon phase information for all the recording dates separately. How to incorporate it in the LDFCs?

**Comment by atruskie** [comment]
So we removed the feature of sundials because ingesting user-generated data is quite complex. There are hundreds of formats a use might come up with and we only supported one fairly esoteric format. It added a lot of complexity to our code that was hard to maintain.

We do have a solution, one not based on user-supplied data, but we haven't had time to implement it.

For now, I'd recommend making the plot yourself. You can do so in R (or any other general-purpose programming language). Plot the graph however you like and make sure it is the same width as the FCS. Then you can stack it on the bottom of the image - that is actually what we do internally.

**Comment by Akash10987** [comment]
> 
> 
> So we removed the feature of sundials because ingesting user-generated data is quite complex. There are hundreds of formats a use might come up with and we only supported one fairly esoteric format. It added a lot of complexity to our code that was hard to maintain.
> 
> We do have a solution, one not based on user-supplied data, but we haven't had time to implement it.
> 
> For now, I'd recommend making the plot yourself. You can do so in R (or any other general-purpose programming language). Plot the graph however you like and make sure it is the same width as the FCS. Then you can stack it on the bottom of the image - that is actually what we do internally.

Thank you, Anthony. I will try this.
